### DISCUSSION TITLE
Template string weirdness inside typealiases

### DISCUSSION THREAD

**Discussion by joshqou** [apple/pkl#377] [Q&A] [answered]
I want to be able to share complex regex groupings with multiple typealiases, like this:

```pkl
hidden ipv6_pattern #"..."#
typealias IPv6Address = String(matches(Regex(#"^\#(ipv6_pattern)$"#)))
```

This typealias will work perfectly fine within the module it is defined in. However, when I import the module containing this line in another pkl file, I get an error about `ipv6_pattern` not being defined:

```
–– Pkl Error ––
Cannot find property `ipv6_pattern`.

17 | typealias IPv6Address = String(matches(Regex(#"^\#(ipv6_pattern)$"#)))
                                                        ^^^^^^^^^^^^
```

This error can be fixed by prefixing `ipv6_pattern` with the module's import name, as typealiases seem to function as macros, but this is very confusing and will break if the import name is changed from the default.

Is there a better way of achieving this? Is this intentional behavior?

This is not specific to matching IP addresses. If pkl has a builtin that provides an IPAddress typealias then that's nice, however I do have other places where I would like to reuse regex snippets too :)

**Comment by bioball** [comment, marked as answer]
This is a bug that will be fixed in 0.26 (fix in https://github.com/apple/pkl/pull/144).

For now, the workaround is to qualify the reference like you mentioned.

Also, there are some helpful typealiases that you can use in [pkl.experimental.net](https://pkl-lang.org/package-docs/pkg.pkl-lang.org/pkl-pantry/pkl.experimental.net/current/index.html), including [IPAddressString](https://pkl-lang.org/package-docs/pkg.pkl-lang.org/pkl-pantry/pkl.experimental.net/current/net/index.html#IPAddressString)

**Reply by joshqou** [reply]
Oh nice
